From 4cb91763a53d827d8ce8596f8ca45356676b1d37 Mon Sep 17 00:00:00 2001
From: Zidan Wang <zidan.wang@freescale.com>
Date: Tue, 28 Jul 2015 09:48:43 +0800
Subject: [PATCH 0333/5242] MLK-11291 radio-si476x: initialize local variable
 rangelow and rangehigh before using them

commit  f1a5f3dfef3e034cc21e546948e45bf665fe1ab4 from
https://source.codeaurora.org/external/imx/linux-imx.git

When transfer rangelow and rangehigh from user space, the local variable
rangelow and rangehigh will not be initialized before using them.

Signed-off-by: Zidan Wang <zidan.wang@freescale.com>
(cherry picked from commit c326f56005f0ee3d0c1fd6ebdebabda919c278dd)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/media/radio/radio-si476x.c |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/media/radio/radio-si476x.c b/drivers/media/radio/radio-si476x.c
index b52e678..d3609de 100644
--- a/drivers/media/radio/radio-si476x.c
+++ b/drivers/media/radio/radio-si476x.c
@@ -778,7 +778,8 @@ static int si476x_radio_s_hw_freq_seek(struct file *file, void *priv,
 		if (err)
 			goto unlock;
 		rangelow = si476x_to_v4l2(radio->core, rangelow);
-	}
+	} else
+		rangelow = seek->rangelow;
 	if (!rangehigh) {
 		err = regmap_read(radio->core->regmap,
 				  SI476X_PROP_SEEK_BAND_TOP,
@@ -786,7 +787,8 @@ static int si476x_radio_s_hw_freq_seek(struct file *file, void *priv,
 		if (err)
 			goto unlock;
 		rangehigh = si476x_to_v4l2(radio->core, rangehigh);
-	}
+	} else
+		rangehigh = seek->rangehigh;
 
 	if (rangelow > rangehigh) {
 		err = -EINVAL;
-- 
1.7.9.5

